source/common/protobuf/utility.cc

```diff
@@ -208,7 +208,7 @@ void ProtoExceptionUtil::throwMissingFieldException(const std::string& field_nam
 void ProtoExceptionUtil::throwProtoValidationException(const std::string& validation_error,
                                                        const Protobuf::Message& message) {
   std::string error = fmt::format("Proto constraint validation failed ({}): {}", validation_error,
-                                  message.DebugString());
+                                  message.GetTypeName());
```

Comment:
Have you considered using absl::StrCat(message) here instead?

Reply:
Yeah I have tried using `absl::StrCat()` but that didn't work. Seems it's because both the `AbslStringify` and the `DebugString` implementation call the [printer.PrintToString](https://github.com/protocolbuffers/protobuf/blob/d91116176642f24ef1b2c2431e7c5c3150bbad68/src/google/protobuf/text_format.cc#L133) where it may crash on certain malformed proto messages.

Reply:
Maybe another way is to remove the param `message` for function `throwProtoValidationException` since GetTypeName is just a workaround to not call DebugString which may crash on malformed input.

Checking input at caller site may not be ideal since the function `throwProtoValidationException` should work for all kinds of input. The root cause here is that protobuf C++ implementation doesn't use exceptions while Envoy uses exceptions, so a call to protobuf C++ code (DebugString here) is not guaranteed to be crash-free and should be avoided to be able to throw the envoy exception successfully no matter what the input is.